### PR TITLE
[Chore]: publish `@fluentui/react-tree-preview` on public docsite + bufixes

### PR DESCRIPTION
**Pull request by bsunderhus**
## New Behavior

1. updates stories according to https://github.com/microsoft/fluentui/pull/28757
2. updates stories with more detailed documentation
3. Adds `@fluentui/react-tree-preview` to public docsite
4. fix: `aria-checked` value to include mixed state
5. fix: `aria-selected` is a required attribute  https://www.w3.org/TR/wai-aria-1.1/#treeitem
6. fix: stops scrolling when pressing Home and End

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4eec212b5109122d7201ef6c5d6cf3c78baf5f8f'>4eec212b5109122d7201ef6c5d6cf3c78baf5f8f</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by layershifter** [comment]
> 1. updates stories according to [Docs(react-tree): Update stories and remove third-party API calls  #28757](https://github.com/microsoft/fluentui/pull/28757)

@bsunderhus not sure that I understand what you meant.

File: packages/react-components/react-tree-preview/stories/Tree/TreeSelection.stories.tsx [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-  FlatTree as Tree,
+  FlatTree,
   TreeItem,
   TreeItemLayout,
   HeadlessFlatTreeItemProps,
   useHeadlessFlatTree_unstable,
 } from '@fluentui/react-tree-preview';
 
-const SELECTION_MODE = 'single'; // change to "multiselect" for multiselection
+const SELECTION_MODE = 'multiselect'; // change to "multiselect" for multiselection
```
Comment (layershifter) [inline packages/react-components/react-tree-preview/stories/Tree/TreeSelection.stories.tsx:10]:
```suggestion
const SELECTION_MODE = 'multiselect';
```

The comment does not make sense

File: packages/react-components/react-tree-preview/stories/Tree/TreeManipulation.stories.tsx [hunk 2]
```diff
@@ -26,23 +26,21 @@ const defaultSubTrees: ItemProps[][] = [
...
-  onRemoveItem: (id: string) => void;
-  isItemRemovable: boolean;
-}) => {
-  const { content, ...treeItemProps } = item.getTreeItemProps();
+type CustomTreeItemProps = TreeItemProps & {
+  onRemoveItem?: (value: string) => void;
+};
+
+const CustomTreeItem = ({ onRemoveItem, ...props }: CustomTreeItemProps) => {
+  const level = props['aria-level'] as number;
```
Comment (layershifter) [inline packages/react-components/react-tree-preview/stories/Tree/TreeManipulation.stories.tsx:34]:
```suggestion
  const level = Number(props['aria-level']);
```

let's not use cast if we can avoid it

File: packages/react-components/react-tree-preview/stories/Tree/TreeSelection.stories.tsx [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 
 type Item = HeadlessFlatTreeItemProps & { content: string };
```
Comment (layershifter) [inline packages/react-components/react-tree-preview/stories/Tree/TreeSelection.stories.tsx:12]: nit: `CustomItem`?

File: packages/react-components/react-tree-preview/stories/Tree/TreeInfiniteScrolling.stories.tsx [hunk 4]
```diff
@@ -23,6 +18,12 @@ const pinnedItems = [
   { value: 'pinned-item-3', parentValue: 'pinned', name: 'Pinned item 3' },
 ];
 
+interface Result {
```
Comment (layershifter) [inline packages/react-components/react-tree-preview/stories/Tree/TreeInfiniteScrolling.stories.tsx:21]:
```suggestion
interface FetchResult {
```

?

**Comment by bsunderhus** [comment]
> > 1. updates stories according to [Docs(react-tree): Update stories and remove third-party API calls  #28757](https://github.com/microsoft/fluentui/pull/28757)
> 
> @bsunderhus not sure that I understand what you meant.

So, now we got to packages, `react-tree` and `react-tree-preview` (yeah, it sucks). this PR basically copy pastes the content from one package into another

File: packages/react-components/react-tree-preview/stories/Tree/TreeActions.stories.tsx [hunk 5]
```diff
@@ -1,3 +1,8 @@
+/** @jsxRuntime classic */
+/** @jsx createElement */
+/** @jsxFrag React.Fragment */
```
Comment (layershifter) [inline packages/react-components/react-tree-preview/stories/Tree/TreeActions.stories.tsx:3]: I don't think that it's a good to use slots for an example. Do customers need it for their use case? Can we avoid their usage in this story?

**Review by layershifter — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by marcosmoura — APPROVED** [review]

**Pull request merged**